File: backend/src/main/java/ma/zs/alc/service/facade/admin/faq/FaqEtudiantAdminService.java
package ma.zs.alc.service.facade.admin.faq;

import java.util.List;
import ma.zs.alc.bean.core.faq.FaqEtudiant;
import ma.zs.alc.dao.criteria.core.faq.FaqEtudiantCriteria;
import ma.zs.alc.zynerator.service.IService;



public interface FaqEtudiantAdminService extends  IService<FaqEtudiant,FaqEtudiantCriteria>  {

    List<FaqEtudiant> findByEtudiantId(Long id);
    int deleteByEtudiantId(Long id);
    long countByEtudiantId(Long id);
    List<FaqEtudiant> findByFaqTypeId(Long id);
    int deleteByFaqTypeId(Long id);
    long countByFaqTypeId(Long id);




}

File: frontend-bqckup/src/app/shared/service/admin/faq/FaqEtudiantAdmin.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';

import {environment} from '../../../../../environments/environment';

import {FaqEtudiantDto} from '../../../model/faq/FaqEtudiant.model';
import {FaqEtudiantCriteria} from '../../../criteria/faq/FaqEtudiantCriteria.model';
import {AbstractService} from "../../../../zynerator/service/AbstractService";


@Injectable({
    providedIn: 'root'
})
export class FaqEtudiantAdminService extends AbstractService<FaqEtudiantDto, FaqEtudiantCriteria> {
    constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    get API() {
        return environment.apiUrlAlcservice + 'admin/faqEtudiant/';
    }

    public constrcutDto(): FaqEtudiantDto {
        return new FaqEtudiantDto();
    }

    public constrcutCriteria(): FaqEtudiantCriteria {
        return new FaqEtudiantCriteria();
    }
}
